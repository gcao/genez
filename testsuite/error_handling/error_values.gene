# Test different types of values can be thrown
(try
  (throw 42)
  (catch e
    (println "Caught number:" e)))

(try  
  (throw true)
  (catch e
    (println "Caught bool:" e)))

(try
  (throw [1 2 3])
  (catch e
    (println "Caught array:" e)))