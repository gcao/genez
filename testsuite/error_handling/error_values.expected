Caught number: 42
Caught bool: true
Caught array: [1 2 3]